src/resolvers/user.ts

```diff
@@ -94,71 +96,77 @@ const resolver: Resolvers = {
       const auth = await getUser();
       return auth;
     },
-    users: async (_, args, { verifyUser, models }): Promise<User[]> => {
+    users: async (
+      _,
+      args,
+      { verifyUser, models },
+    ): Promise<UsersConnection> => {
       const { User: userModel } = models;
       const auth = verifyUser();
       checkAuth(auth);
 
-      const { user, includeUser, filter, first, after } = args;
+      const { filter, user, includeUser, pageSize = 20, after } = args;
+      let qryUser: object = {};
+      let qryAfter: object = {};
+      let qryIncludeUser: object = {};
 
-      let query: object = {};
-      if (after) {
-        query = {
-          id: { [Op.gt]: after },
+      if (filter && user) {
+        const userOpConds: object = {};
+        for (const col in user) {
+          userOpConds[col] = {
+            [Op.like]: `%${user[col]}%`,
+          };
+        }
+        qryUser = {
+          [Op.or]: userOpConds,
+        };
+      } else if (user) {
+        qryUser = {
+          ...user,
         };
       }
-
-      let limit: number;
-      if (first) {
-        limit = first;
-      }
-
-      if (includeUser === false) {
-        return userModel.findAll({
-          where: {
-            ...user,
-            ...query,
-            id: {
-              [Op.ne]: auth.userId,
-            },
-            verified: true,
+      if (includeUser) {
+        qryIncludeUser = {
+          id: {
+            [Op.ne]: auth.userId,
           },
-          limit,
-          order: [
-            ['id', 'ASC'],
-          ],
-        });
+        };
+      }
+      if (after) {
+        qryAfter = {
+          createdAt: { [Op.lt]: Number(after) },
+        };
       }
+      const where: WhereOptions = {
+        ...qryUser,
+        ...qryAfter,
+        ...qryIncludeUser,
```

Comment:
It'd be better if you have an idea of making these variables into one. Maybe just `query`. It is hard to imagine at a glance what `qryUser`,  `qryAfter`, `qryIncludeUser` are.